src/java/org/apache/cassandra/index/sai/disk/vector/JVectorLuceneOnDiskGraph.java

```diff
@@ -65,6 +69,8 @@ protected JVectorLuceneOnDiskGraph(SegmentMetadata.ComponentMetadataMap componen
      */
     public abstract VectorPostingList search(float[] queryVector, int topK, float threshold, int limit, Bits bits, QueryContext context);
 
+    public abstract ScoredRowIdIterator searchWithScores(float[] queryVector, int topK, Bits bits, QueryContext context);
```

Comment:
as mentioned above I'd like to combine search and searchWithScores to return an intermediate result like RowIdIterator that can be transformed to VPL or SRII by caller